src/main/scala/mesosphere/marathon/core/task/bus/TaskStatusObservables.scala

```diff
@@ -9,13 +9,17 @@ object TaskStatusObservables {
   case class TaskStatusUpdate(timestamp: Timestamp, taskId: Task.Id, status: MarathonTaskStatus) {
     def appId: PathId = taskId.appId
   }
+  case class TaskUpdate(stateOp: TaskStateOp, stateChange: TaskStateChange) {
+    def taskId: Task.Id = stateOp.taskId
+    def appId: PathId = stateOp.taskId.appId
+  }
 }
 
 /**
   * Allows subscription to TaskStatus updates.
   */
 trait TaskStatusObservables {
-  def forAll: Observable[TaskStatusUpdate]
-  def forAppId(appId: PathId): Observable[TaskStatusUpdate]
+  def forAll: Observable[TaskUpdate]
```

Comment:
With this change you actually already moved pretty close to remove these TaskStatusObervables alltogether. Nice!